.github/workflows/tools.yml

```diff
@@ -7,105 +7,133 @@ on:
   push:
     branches: [ master ]
     paths:
-      - tools/**
+      - 'tools/**'
   pull_request:
     branches: [ master ]
     paths:
-      - tools/**
+      - 'tools/**'
 
 jobs:
   build:
+    name: [ master-runner ]
     runs-on: ${{ matrix.os }}
     timeout-minutes: 60
     strategy:
       matrix:
         os: [macos-latest, ubuntu-latest, windows-latest]
     steps:
     - uses: actions/checkout@v2
+    - uses: dorny/paths-filter@v2
+      id: changes
+      with:
+        filters: |
+          abel:
+            - 'tools/abel/**'
+          explorer:
+            - 'tools/explorer/**'
+          guid_generator:
+            - 'tools/guid_generator/**'
+          instance_validator:
+            - 'tools/validators/instance_validator/**'
+          rdf_generator:
+            - 'tools/rdf_generator/**'
+          type_validator:
+            - 'tools/validators/ontology_validator/**'
+          scoring:
+            - 'tools/scoring/**'
     - name: Set up Python 3.9
       uses: actions/setup-python@v1
       with:
         python-version: 3.9
     - name: Install Pylint
       run: |
         python -m pip install pylint
+    - name: Run Pylint on tools
+      run: |
+        pylint --rcfile ./styles/pylintrc ./tools
+
 
      #TODO(b/210730979): Investigate calling ontology validator setup with instance validator setup
      #---------- Ontology Explorer ----------#
-    - name: Ontology Validator Install dependencies
-      run: |
-        python setup.py install
-      working-directory: ./tools/validators/ontology_validator
-    - name: Instance Validator Install dependencies
-      run: |
-        python -m pip install .
-      working-directory: ./tools/validators/instance_validator
     - name: Ontology Explorer Install dependencies
+      if: steps.changes.outputs.explorer == 'true'
       run: |
         python setup.py install
       working-directory: ./tools/explorer
-    - name: Run Pylint on Ontology Explorer
-      run: |
-        pylint --rcfile ./styles/pylintrc ./tools/explorer
     - name: Ontology Explorer Tests
```

Comment:
this will break though? it will not find the ontology type validator dependencies

Reply:
I fixed the order of operations.